balances/src/lib.rs

```diff
@@ -251,21 +250,23 @@ impl<T: Config> Pallet<T> {
 		demurrage: BalanceType,
 	) -> BalanceEntry<T::BlockNumber> {
 		let current_block = frame_system::Pallet::<T>::block_number();
-		let elapsed_time_block_number = current_block - entry.last_update;
-		let elapsed_time_u32: u32 = elapsed_time_block_number
-			.try_into()
-			.ok()
-			.expect("blockchain will not exceed 2^32 blocks; qed");
-		let elapsed_time = BalanceType::from_num(elapsed_time_u32);
-		let exponent: BalanceType = -demurrage * elapsed_time;
-		let exp_result: BalanceType = exp(exponent).unwrap();
```

Comment:
This panicked because the `exp` function computes the result by flipping the negative sign, and inverting the result at the end again.